MapboxNavigation/UserHaloCourseView.swift

```diff
@@ -0,0 +1,117 @@
+import UIKit
+import Turf
+import Mapbox
+
+/**
+A view representing the user’s reduced accuracy location on screen.
+*/
+
+public class UserHaloCourseView: UIView, CourseUpdatable {
+    private var lastLocationUpdate: Date?
+
+    /**
+    Transforms the location of the user puck.
+    */
+    public func update(location: CLLocation, pitch: CGFloat, direction: CLLocationDegrees, animated: Bool, tracksUserCourse: Bool) {
```

Comment:
I think we need it. Because when the location accuracy changes, the user course view switches between `UserPuckCourseView` and `UserHaloCourseView` . Both of them follow the same protocol of `CourseUpdatable`, which has the `update` function to support the location and pitch change during the navigation.